### PR TITLE
[feat] Add new build system backend for custom builds

### LINKED ISSUE #2473: Add a minimal `Custom` build system
This should allow users to build their code with a custom command. It should not do any interpretation of the current environ. This feature should be used with caution, since environment management, reproducibility and any potential side effects are all controlled by the user's custom build system.

### PR DESCRIPTION
**Pull request by vkarak**
Fixes #2473

### CODE REVIEW

**Review by ekouts — APPROVED** [review]
lgtm

File: reframe/core/buildsystems.py [hunk 1]
```diff
@@ -897,6 +897,29 @@ def prepare_cmds(self):
         return cmds
 
 
+class CustomBuild(BuildSystem):
```
Comment (teojgo) [inline reframe/core/buildsystems.py:900]: I would rename this to `Custom` since the user is going to use:
```
build_system = 'Custom'
```

**Review by teojgo — CHANGES_REQUESTED** [review]
Since you are touching the buildsystems, can you change the remaining string formatting which uses `%` to f=strings?

File: reframe/core/buildsystems.py (hunk 1 above)
Comment (vkarak) [reply reframe/core/buildsystems.py:900]: Yeah, I though about this, but `build_system = 'CustomBuild'` is not bad either, is it? In the sense that you are doing "custom build" after all.

Comment (teojgo) [reply reframe/core/buildsystems.py:900]: I am fine with both.

**Review by teojgo — APPROVED** [review]

**Pull request merged**